src/decisionengine/framework/engine/Workers.py

```diff
@@ -51,60 +50,42 @@ def get_consumes(self):
         return self.task_manager.get_consumes()
 
     def run(self):
-
         myname = self.task_manager.name
         myfilename = os.path.join(os.path.dirname(self.logger_config["log_file"]), myname + ".log")
+        unit_testing = self.logger_config.get("unit_testing", "False")
 
-        self.logger = structlog.getLogger(CHANNELLOGGERNAME)
+        self.logger = structlog.getLogger(logconf.CHANNELLOGGERNAME)
 
         # setting a default value here. value from config file is set in call
         # self.task_manager.set_loglevel_value after logger configuration is completed
-        logging.getLogger(CHANNELLOGGERNAME).setLevel(logging.DEBUG)
+        logging.getLogger(logconf.CHANNELLOGGERNAME).setLevel(logging.DEBUG)
 
-        # TODO:
-        # alter decisionengine.framework.modules.de_logger set_logging
-        # so we can reuse it here and reduce duplication
         logger_rotate_by = self.logger_config.get("file_rotate_by", "size")
-
-        # logconf is our global object edited in global space on purpose
-
         if logger_rotate_by == "size":
-            logconf["handlers"].update(
-                {
-                    myname: {
-                        "level": "DEBUG",
-                        "filename": myfilename,
-                        "formatter": "plain",
-                        "class": "logging.handlers.RotatingFileHandler",
-                        "maxBytes": MAX_CHANNEL_FILE_SIZE,
-                        "backupCount": self.logger_config.get("max_backup_count", 6),
-                    }
-                }
+            handler = logging.handlers.RotatingFileHandler(
+                filename=myfilename,
+                maxBytes=self.logger_config.get("max_file_size", 200 * MB),
+                backupCount=self.logger_config.get("max_backup_count", 6),
             )
+
         elif logger_rotate_by == "time":
-            logconf["handlers"].update(
-                {
-                    myname: {
-                        "level": "DEBUG",
-                        "filename": myfilename,
-                        "formatter": "plain",
-                        "class": "logging.handlers.TimedRotatingFileHandler",
-                        "when": "D",
-                        "interval": 1,
-                    }
-                }
+            handler = logging.handlers.TimedRotatingFileHandler(
```

Comment:
Tests don't seem to be covering the TimedRotatingFileHandler

Reply:
Will add that.

Reply:
I have added a unit test for this as requested.